guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,395 @@
+"""
+Title: CutMix, MixUp, and RandAugment image augmentation with KerasCV
+Author: [lukewood](https://lukewood.xyz)
+Date created: 2022/04/08
+Last modified: 2022/04/08
+Description: Use KerasCV to augment images with CutMix, MixUp, RandAugment, and more!
+"""
+
+"""
+## Overview
+
+KerasCV makes it easy to assemble state-of-the-art, industry-grade data augmentation
+pipelines for image classification and object detection tasks. KerasCV offers a wide
+suite of preprocessing layers implementing common data augmentation techniques.
+
+Perhaps three of the most useful layers are `CutMix`, `MixUp`, and `RandAugment`. These
+layers are used in nearly all state-of-the-art image classification pipelines.
+
+This guide will show you how to compose these layers into your own data
+augmentation pipeline for image classification tasks. This guide will also walk you
+through the process of customizing a KerasCV data augmentation pipeline.
+"""
+
+"""
+## Imports & setup
+
+This tutorial requires you to have KerasCV installed:
+
+```shell
+pip install keras-cv
+```
+
+We begin by importing all required packages:
+"""
+
+import keras_cv
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow import keras
+from tensorflow.keras import applications
+from tensorflow.keras import losses
+from tensorflow.keras import optimizers
+
+"""
+## Data loading
+
+This guide uses the
+[102 Category Flower Dataset](https://www.robots.ox.ac.uk/~vgg/data/flowers/102/)
+for demonstration purposes.
+
+To get started, we first load the dataset:
+"""
+
+BATCH_SIZE = 32
+AUTOTUNE = tf.data.AUTOTUNE
+tfds.disable_progress_bar()
+data, dataset_info = tfds.load("oxford_flowers102", with_info=True, as_supervised=True)
+steps_per_epoch = dataset_info.splits["train"].num_examples // BATCH_SIZE
+
+"""
+Next, we resize the images to a constant size, `(224, 224)`, and one-hot encode the
+labels. Please note that `keras_cv.layers.CutMix` and `keras_cv.layers.MixUp` expect
+targets to be one-hot encoded. This is because they modify the values of the targets
+in a way that is not possible with a sparse label representation.
+"""
+
+IMAGE_SIZE = (224, 224)
+num_classes = dataset_info.features["label"].num_classes
+
+
+def prepare(image, label):
+    image = tf.image.resize(image, IMAGE_SIZE)
+    image = tf.cast(image, tf.float32)
+    label = tf.one_hot(label, num_classes)
+    return {"images": image, "labels": label}
+
+
+def prepare_dataset(dataset, split):
+    if split == "train":
+        return (
+            dataset.map(prepare, num_parallel_calls=AUTOTUNE)
+            .batch(BATCH_SIZE)
+            .shuffle(10 * BATCH_SIZE)
+        )
+    if split == "test":
+        return dataset.map(
+            lambda x, y: prepare(x, y), num_parallel_calls=AUTOTUNE
```

Comment:
Here `dataset.map` on `prepare` function coded differently for the  `train` and `test` split. Either we can use `lambda` or just pass the function.